composer/trainer/trainer.py

```diff
@@ -2686,6 +2696,13 @@ def _eval_loop(
                 rank_num_samples = data_spec.get_num_samples_in_batch(self.state.batch)
                 rank_num_tokens = data_spec.get_num_tokens_in_batch(self.state.batch)
 
+                # If using a distributed sampler, keep track of last_batch for metrics update
+                if dist_sampler is not None and dataset_len is not None and drop_last == False:
+                    batch_num_samples_tensor = self.state.device.tensor_to_device(torch.tensor(rank_num_samples))
+                    dist.all_reduce(batch_num_samples_tensor, reduce_operation='SUM')
+                    batch_num_samples = batch_num_samples_tensor.item()
+                    last_batch = self.state.eval_timestamp.sample + batch_num_samples > dataset_len
```

Comment:
If it's equal, there's no padding so it's fine to ignore. We only want to handle the case where there's padding.

I will update to >= because otherwise the variable is confusing, but I'm not going to add a unit test because it (correctly) no-ops in both > or >=. With >=, This will never pass:
```
padding = dist_sampler.total_size - dataset_len
if dist.get_global_rank() >= dist.get_world_size() - padding:
```